### ISSUE TITLE
Test Suite to verify Pact implementations? 

### ISSUE THREAD

**Issue by bmarini** [pact-foundation/pact-specification#46]
Has anyone considered developing a test suite that you could run against a pact implementation? Similar to https://github.com/ruby/spec, it could be used to verify an implementation conforms to the spec at any version.

**Comment by bmarini** [comment]
So, I just noticed this:
> 4. Do it! Ensure your implementation passes all the tests in the testcases directory of the appropriate branch. Start with version-1 and get that out and battle test it. Then, try for versions 2 and 3.

I'm not sure I understand how to verify against those testcast json files, is that just up to the implementor to decide?

**Comment by bethesque** [comment]
Here's an example of how the ruby compliance tests are done: https://github.com/realestate-com-au/pact/blob/master/spec/pact_specification/compliance-1.0.0.rb

**Comment by bmarini** [comment]
Cool thanks!

**Comment by YOU54F** [comment]
Yes this is on our mind, we would love to get some additional help and input on it and we could see this being valuable

cc: @uglyog / @mefellows / @mikegeeves 

https://github.com/mikegeeves/pact-examples
https://github.com/pact-foundation/pact-compatibility-suite
